pyamg/aggregation/aggregation.py

```diff
@@ -227,18 +227,34 @@ def smoothed_aggregation_solver(A, B=None, BH=None,
 
     if A.shape[0] != A.shape[1]:
         raise ValueError('expected square matrix')
-    # Right near nullspace candidates
+    # Right near nullspace candidates use constant for each variable as default
     if B is None:
-        B = np.ones((A.shape[0], 1), dtype=A.dtype)  # use constant vector
+        B = np.kron(np.ones((int(A.shape[0]/blocksize(A)), 1), dtype=A.dtype),
+                    np.eye(blocksize(A)))
     else:
         B = np.asarray(B, dtype=A.dtype)
+        if len(B.shape) == 1:
+            B = B.reshape(-1, 1)
+        if B.shape[0] != A.shape[0]:
+            raise ValueError('The near null-space modes B have incorrect \
+                              dimensions for matrix A')
+        # if B.shape[1] < blocksize(A):
+        #     raise ValueError('B.shape[1] must be >= the blocksize of A')
```

Comment:
Should we change this to a warning, because this choice can degrade convergence in standard test problems.